### PR TITLE
statistics: recalculate bucket repeats when merging global statistics

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary: statistics: recalculate bucket repeats when merging global statistics

### What is changed and how it works?

statistics: recalculate bucket repeats when merging global statistics

The upper is the result of this PR while the lower is Mater.
We can see after recalculating bucket repeats, `max-perrors` decrease sharply. 

image

### Check List 

Tests 

- Unit test

### Release note 

- statistics: recalculate bucket repeats when merging global statistics

### CODE REVIEW

**Review by Reminiscent — APPROVED** [review]
/LGTM

File: statistics/histogram.go [hunk 1]
```diff
@@ -1919,6 +1919,18 @@ func MergePartitionHist2GlobalHist(sc *stmtctx.StatementContext, hists []*Histog
 		globalBuckets[i].lower = globalBuckets[i-1].upper.Clone()
 		globalBuckets[i].Count = globalBuckets[i].Count + globalBuckets[i-1].Count
 	}
+
+	// Recalculate repeats
+	for _, bucket := range globalBuckets {
```
Comment (rebelice) [inline statistics/histogram.go:1924]: This is not the fastest implementation. We may add comments to remind us to optimize performance later.

Comment (qw4990) [reply statistics/histogram.go:1925]: A comment is added

**Review by rebelice — APPROVED** [review]

**Pull request merged**